src/cpp/database/database.cpp

```diff
@@ -912,6 +912,20 @@ std::vector<std::pair<EntityId, EntityId>> Database::get_entities_by_guid(
     throw Unsupported("Not implemented yet");
 }
 
+EntityKind Database::get_entity_kind(
+        EntityId entity_id) const
+{
+    try
+    {
+        auto entity = get_entity(entity_id);
+        return entity.get()->kind;
+    }
+    catch (Exception& BadParameter)
+    {
+        return EntityKind::INVALID;
+    }
```

Comment:
I'd actually let the exception to be thrown from `get_entity`, why would we need to do otherwise?